InterGalacticCurrencyConvertor/src/main/java/numerals/RomanNumerals.java

```diff
@@ -0,0 +1,145 @@
+package main.java.numerals;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+
+/**
+ * Keeps track of Roman Numerals, its validation and decimal conversion.
+ */
+public class RomanNumerals {
+	private HashMap<String, Integer> mMap;
+	private static RomanNumerals mInstance = new RomanNumerals();
+	private RomanNumerals() {
+		mMap = new HashMap<>();
+		mMap.put("i", 1);
+		mMap.put("v", 5);
+		mMap.put("x", 10);
+		mMap.put("l", 50);
+		mMap.put("c", 100);
+		mMap.put("d", 500);
+		mMap.put("m", 1000);
+	}
+	public static RomanNumerals getInstance() {
+		return mInstance;
+	}
+	public boolean validate(ArrayList<String> inRNums) {
+		boolean vResult = false;
+		if (inRNums == null || inRNums.size() == 0) {
+			return vResult;
+		}
+		HashMap<String, Integer> vOccurrenceMap = new HashMap<>();
+		for (String rn : mMap.keySet()) {
+			vOccurrenceMap.put(rn, 0);
+		}
+		if (!isRomanNumeral(inRNums.get(0))) {
+			return vResult;
+		}
+		vOccurrenceMap.put(inRNums.get(0), vOccurrenceMap.get(inRNums.get(0)) + 1);
+		int vBefore = getDecimalVal(inRNums.get(0)), vAfter = 0, vCurRepRNCnt = 1;
```

Comment:
Add comments to various blocks of the validate() method since it is too big.

Also, what does the prefix 'v' mean before each local variable? Consider removing such prefixes. They are not generally used in Java (except maybe for the class member variables)

Reply:
updated